rust/main/agents/relayer/src/server/list_messages.rs

```diff
@@ -108,15 +123,40 @@ mod tests {
     #[tokio::test]
     async fn test_message_id_retry() {
         let (addr, op_queue) = setup_test_server();
-        let dummy_operation_1 =
-            Box::new(MockPendingOperation::new(1, DUMMY_DOMAIN.into())) as QueueOperation;
-        let dummy_operation_2 =
-            Box::new(MockPendingOperation::new(2, DUMMY_DOMAIN.into())) as QueueOperation;
-        let v = vec![
-            serde_json::to_value(&dummy_operation_1).unwrap(),
-            serde_json::to_value(&dummy_operation_2).unwrap(),
-        ];
-        let expected_response = serde_json::to_string_pretty(&v).unwrap();
+        let id_1 = "0x1acbee9798118b11ebef0d94b0a2936eafd58e3bfab91b05da875825c4a1c39b";
+        let id_2 = "0x51e7be221ce90a49dee46ca0d0270c48d338a7b9d85c2a89d83fac0816571914";
+        let dummy_operation_1 = Box::new(
+            MockPendingOperation::new(1, DUMMY_DOMAIN.into())
+                .with_id(H256::from_str(id_1).unwrap()),
+        ) as QueueOperation;
+        let dummy_operation_2 = Box::new(
+            MockPendingOperation::new(2, DUMMY_DOMAIN.into())
+                .with_id(H256::from_str(id_2).unwrap()),
+        ) as QueueOperation;
+        let expected_response = r#"[
+  {
+    "id": "0x1acbee9798118b11ebef0d94b0a2936eafd58e3bfab91b05da875825c4a1c39b",
+    "operation": {
+      "destination_domain": {
+        "Known": "Arbitrum"
+      },
+      "id": "0x1acbee9798118b11ebef0d94b0a2936eafd58e3bfab91b05da875825c4a1c39b",
```

Comment:
the reason there already is an `id` in here is because it's a field on `MockOperation` - that field is missing on `HyperlaneMessage` because it's derived, so now we'll have it explicit

Reply:
makes sense, a comment may be useful but nbd